packages/flutter/lib/src/painting/box_border.dart

```diff
@@ -538,15 +578,25 @@ class Border extends BoxBorder {
       }
     }
 
+    // Allow painting non-uniform borders if the color is uniform.
```

Comment:
```suggestion
    // Allow painting non-uniform borders if the shape is rectangular, and the color and style are uniform.
```

Reply:
I think I'll allow BoxShape.circle. I've been trying to deprecate it for 8 months, but that will give a reason for it to exist.

Let me see if I can implement non-uniform border for circles.